pytorch_lightning/callbacks/device_stats_monitor.py

```diff
@@ -43,17 +47,21 @@ class DeviceStatsMonitor(Callback):
         >>> trainer = Trainer(callbacks=[device_stats]) # doctest: +SKIP
     """
 
+    def __init__(self, cpu_stats: bool = True) -> None:
+        super().__init__()
+        self.cpu_stats = cpu_stats
+
     def setup(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", stage: Optional[str] = None) -> None:
         if not trainer.logger:
             raise MisconfigurationException("Cannot use DeviceStatsMonitor callback with Trainer that has no logger.")
 
-    def on_train_batch_start(
+    def _on_train_batch_common(
```

Comment:
Why not keep on_train_batch_start and on_train_batch_end?

Reply:
The functions we're doing the same thing. Seemed like a lot of duplicated code

Reply:
I see. Personally I prefer it how it was previously for clarity and readability. What do others thinks?

Reply:
@daniellepintz I'm marking this as resolved after making @tchaton 's suggested change, but open to discussing further if you wish